src/contracts/pods/EigenPod.sol

```diff
@@ -239,7 +252,7 @@ contract EigenPod is IEigenPod, Initializable, ReentrancyGuardUpgradeable, Eigen
      * @param validatorFields are the fields of the "Validator Container", refer to consensus specs
      * @dev For more details on the Beacon Chain spec, see: https://github.com/ethereum/consensus-specs/blob/dev/specs/phase0/beacon-chain.md#validator
      */
-    function verifyOvercommittedStake(
+    function verifyBalanceUpdate(
         uint40 validatorIndex,
         BeaconChainProofs.ValidatorFieldsAndBalanceProofs calldata proofs,
         bytes32[] calldata validatorFields,
```

Comment:
almost feel like we should create a library for a `Validator` struct with functions like `Validator.getWithdrawalCredentials()` and that way when the beacon spec changes we don't change this file at all, only the libraries

Reply:
Oh I like that idea.... do all the proof specific index calculations, etc. separately.